azurecompute/src/main/java/org/jclouds/azurecompute/compute/AzureComputeServiceAdapter.java

```diff
@@ -267,6 +267,19 @@ public boolean apply(final OSImage input) {
 
    @Override
    public Deployment getNode(final String id) {
+      final Deployment deployment = api.getDeploymentApiForService(id).get(id);
+      if (deployment != null && !deployment.roleInstanceList().isEmpty()) {
```

Comment:
This conditional is the same one than the used below. Refactor this a bit to reuse the code.